CARDS-2512: PM Acute Care methodology

sdumitriu opened the pull request:
- Build with `mvn clean install -Pdocker`
- In `compose-cluster` make a docker-compose file with `python3 generate_compose_yaml.py --mssql --cards_project cards4prems --oak_filesystem --dev_docker_image --composum --adminer --smtps --smtps_test_container --smtps_test_mail_path ~/cards-mail --server_address localhost:8080` then start it up with `docker-compose build && docker-compose up`
- At `http://localhost:1435/?mssql=mssql&username=sa&db=master&ns=path&import=` import the attached SQL files, [sqls.tar.gz](https://github.com/user-attachments/files/16970864/sqls.tar.gz) (unzip the archive first to get 3 sql files)
- Log in to cards at `http://localhost:8080/`
- Import YE visits: `http://localhost:8080/Subjects.importClarity?dayToQuery=-7&config=Your%20Experience%20-%20Inpatient%20and%20ED%20discharge%20events`
- Check that visits for patients 1, 2, and 3 were imported
- Import YVM visits: `http://localhost:8080/Subjects.importClarity?dayToQuery=-2&config=Your%20Voice%20Matters%20-%20PMCC%20outpatient%20visits`

- Check that YVM visits for patients 11, 12, 13, 21 and 29 were imported
- Import AC visits: `http://localhost:8080/Subjects.importClarity?dayToQuery=-7&config=PMH%20Acute%20Care%20visits`
- Check that AC visits for patients 21, 22 and 23 were imported; make sure 21 has TWO visits, one for YVM and one for AC
- "Import" AC discard visits: `http://localhost:8080/Subjects.importClarity?dayToQuery=-2&config=PMH%20AC%20Trumps%20YVM`
- Check that the YVM visit for 29 was deleted, and the YVM visit for 21 was not deleted
- In the end there should be 10 Visit Information forms, ED for 1&2, IP for 3, YVM for 11, 12, 13, 21, and AC for 21, 22, 23
- Stop docker-compose and clean up with `docker-compose rm -vf && ./cleanup.sh`

sdumitriu committed:
CARDS-2512: PM Acute Care methodology
Add a placeholder Acute Care questionnaire, currently just a copy of OED

sdumitriu committed:
CARDS-2512: PM Acute Care methodology
Add questionnaire sets for AC and AC+IP

sdumitriu committed:
CARDS-2512: PM Acute Care methodology
Add the two new clinics

sdumitriu committed:
CARDS-2512: PM Acute Care methodology
Configure the Clarity import

sdumitriu committed:
CARDS-2512: PM Acute Care methodology
Update test data

sdumitriu committed:
CARDS-2512: PM Acute Care methodology
Added sidebar entries

sdumitriu committed:
CARDS-2512: PM Acute Care methodology
Added CopyAnswers and Statistics (placeholders)

sdumitriu commented on prems-resources/clinical-data/src/main/resources/SLING-INF/content/Questionnaires/PMHAC.xml:
Needs the proper questionnaire, this is just OED.

sdumitriu commented on prems-resources/clinical-data/src/main/resources/SLING-INF/content/Statistics/AC-Overall.xml:
```diff
@@ -0,0 +1,53 @@
+	</property>
+	<property>
+		<name>xVar</name>
+		<value>/Questionnaires/PMHAC/oed_module1/oed_8</value>
```
Update with the proper question path.

sdumitriu commented on prems-resources/clinical-data/src/main/resources/SLING-INF/content/Statistics/AC-Overall.xml:
```diff
@@ -0,0 +1,53 @@
+	</property>
+	<property>
+		<name>splitVar</name>
+		<value>/Questionnaires/PMHAC/oed_module1/oed_hospital</value>
```
Update with the proper question path.

sdumitriu commented on prems-resources/clinical-data/src/main/resources/SLING-INF/content/Survey/AC.xml:
```diff
@@ -0,0 +1,92 @@
+      "link" : "dashboard+path"
+    },
+    {
+      "key": "oed_9",
```
Update with the proper question.

sdumitriu commented on prems-resources/clinical-data/src/main/resources/SLING-INF/content/Survey/ACIP.xml:
```diff
@@ -0,0 +1,140 @@
+      "link" : "dashboard+path"
+    },
+    {
+      "key": "oed_9",
```
Update with the proper question.

sdumitriu commented on prems-resources/clinical-data/src/main/resources/SLING-INF/content/Survey/AC.xml:
```diff
@@ -0,0 +1,92 @@
+        </property>
+        <property>
+            <name>estimate</name>
+            <value>5</value>
```
Update if needed.

sdumitriu commented on prems-resources/clinical-data/src/main/resources/SLING-INF/content/Survey/ACIP.xml:
```diff
@@ -0,0 +1,140 @@
+        </property>
+        <property>
+            <name>estimate</name>
+            <value>5</value>
```
Update if needed.

sdumitriu commented on prems-resources/clinical-data/src/main/resources/SLING-INF/content/Survey/ACIP.xml:
```diff
@@ -0,0 +1,140 @@
+    <property>
+        <name>intro</name>
+        <value>
+Please answer some questions about your recent emergency visit and hospital stay at:
```
Update.

sdumitriu commented on prems-resources/clinical-data/src/main/resources/SLING-INF/content/Survey/AC.xml:
```diff
@@ -0,0 +1,92 @@
+    <property>
+        <name>intro</name>
+        <value>
+Please answer some questions about your recent emergency visit at:
```
Update.

sdumitriu commented on prems-resources/clinical-data/src/main/resources/SLING-INF/content/Survey/ClinicMapping.xml:
```diff
@@ -290,4 +290,52 @@
+        </property>
+        <property>
+            <name>displayName</name>
+            <value>PMH Acute Care and Inpatient</value>
```
Is this right?

sdumitriu commented on prems-resources/clinical-data/src/main/resources/SLING-INF/content/Survey/ClinicMapping.xml:
```diff
@@ -290,4 +290,52 @@
+        </property>
+        <property>
+            <name>displayName</name>
+            <value>PMH Acute Care</value>
```
Is this right?

sdumitriu commented on prems-resources/clinical-data/src/main/resources/SLING-INF/content/apps/cards/config/CopyAnswers/Questionnaires/PMHAC.json:
```diff
@@ -0,0 +1,6 @@
+{
+  "jcr:primaryType": "nt:unstructured",
+  "jcr:reference:oed_9": "/Questionnaires/PMHAC/oed_section/oed_9",
```
Update with the proper question path.

marta- committed:
CARDS-2511: PM Acute Care questionnaire

marta- committed:
CARDS-2512: PM Acute Care methodology
* Renamed the questionnaire
* Corrected the "Overall" question id based on the updated questionnaire structure
* Updated the Statistic config
* Added CopyAnswers config

marta- committed:
CARDS-2512: PM Acute Care Methodology
Minor reordering of side bar entries

sdumitriu commented on prems-resources/clinical-data/src/main/resources/SLING-INF/content/Questionnaires/PMHAC.xml:
```diff
@@ -1043,123 +1038,161 @@
+		</node>
+	</node>
+	<node>
+		<name>oed_module2</name>
```
oed?

sdumitriu commented on prems-resources/clinical-data/src/main/resources/SLING-INF/content/apps/cards/config/CopyAnswers/Questionnaires/PMHAC.json:
This should be renamed.

sdumitriu commented on prems-resources/clinical-data/src/main/resources/SLING-INF/content/apps/cards/config/CopyAnswers/Questionnaires/AC.json:
Shouldn't this be deleted?

marta- committed:
CARDS-2511: PM Acute Care questionnaire
* Fixed the pagination mode
* Minor indentation and wording fixes
* Added hidden reference question for department

marta- committed:
CARDS-2511: PM Acute Care questionnaire
* Reorganized the questions
* Naming cleanup

marta- committed:
CARDS-2512: PM Acute Care Methodology
Renamed PMH to PM in Acute Care configs

marta- committed:
CARDS-2512: PM Acute Care Methodology
Added email templates and schedules

marta- committed:
CARDS-2512: PM Acute Care Methodology
Added export config for PM Acute Care

marta- committed:
CARDS-2542: Make the multi-valued `provider` answer from Visit information available to be displayed in the survey invitation email

marta- committed:
CARDS-2512: PM Acute Care Methodology
Updated email templates according to new specs

marta- committed:
CARDS-2511: PM Acute Care questionnaire
Typo fix

marta- commented:
Rebased on #1789 - emails can now be tested @sashaandjic . The first paragraph of the emails should include the unit visited by the patient, as specified in the `provider` answer of Visit information.

Email schedule: initial - 7 days after visit, reminders  - 14 and 21 days after visit.

sashaandjic commented:
![image](https://github.com/data-team-uhn/cards/assets/23243244/8280a632-7607-45f0-976b-790a1628b36d)

sashaandjic reviewed (approved):
tested and EDITED
I was wrong about and extra`Not applicable` option for question pmac_7_ad_5. It is supposed to be there.

sdumitriu committed:
CARDS-2511: PM Acute Care questionnaire
Updated according to the latest data table and specs

sdumitriu committed:
CARDS-2511: PM Acute Care questionnaire
Discard YVM surveys when an Acute Care survey would be sent for the same day

sdumitriu committed:
CARDS-2511: PM Acute Care questionnaire
Fixed some bugs

sdumitriu committed:
CARDS-2511: PM Acute Care questionnaire
Revert no longer needed changes to the import rules

sdumitriu reviewed (commented):
@marta- One thing missing: we import the **first** visit for the day, not the last one.

sashaandjic reviewed (approved):
tested

![image](https://github.com/user-attachments/assets/bfe355c4-eea3-45be-8c64-10b58fab7678)

![image](https://github.com/user-attachments/assets/2aef4046-9c1c-4901-a8f4-d0eb6e5b70de)

![image](https://github.com/user-attachments/assets/8d4f9655-8c09-41a7-a633-40e93f971a4b)

![image](https://github.com/user-attachments/assets/6d3251e7-89fa-461d-a10b-f0af0334e774)

![image](https://github.com/user-attachments/assets/9645c1a3-1e55-4cde-bdb8-ffb3e8acfc3c)

sashaandjic commented:
AC - second email ( 14 days)
![image](https://github.com/user-attachments/assets/ec0be4d2-e2e7-4df3-bb69-678794015c0e)

sdumitriu committed:
CARDS-2511: PM Acute Care questionnaire
Import the newest event in a day for Acute Care visits

sdumitriu committed:
CARDS-2511: PM Acute Care questionnaire
Don't include the PE logo in emails as attachment, since it is not used in the email body

sashaandjic reviewed (approved):
tested the latest change. only the latest AC visit gets imported.
![image](https://github.com/user-attachments/assets/69ab18b2-b22c-49cd-a282-faced71d8486)
![image](https://github.com/user-attachments/assets/58d0caf5-ae20-4987-86ab-4ab407039475)
